document-store/src/main/java/org/hypertrace/core/documentstore/metric/postgres/PostgresDocStoreMetricProvider.java

```diff
@@ -0,0 +1,82 @@
+package org.hypertrace.core.documentstore.metric.postgres;
+
+import static org.hypertrace.core.documentstore.expression.operators.AggregationOperator.COUNT;
+import static org.hypertrace.core.documentstore.expression.operators.RelationalOperator.EQ;
+import static org.hypertrace.core.documentstore.model.config.CustomMetricConfig.VALUE_KEY;
+
+import java.util.List;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.hypertrace.core.documentstore.expression.impl.AggregateExpression;
+import org.hypertrace.core.documentstore.expression.impl.ConstantExpression;
+import org.hypertrace.core.documentstore.expression.impl.IdentifierExpression;
+import org.hypertrace.core.documentstore.expression.impl.RelationalExpression;
+import org.hypertrace.core.documentstore.metric.BaseDocStoreMetricProviderImpl;
+import org.hypertrace.core.documentstore.metric.DocStoreMetric;
+import org.hypertrace.core.documentstore.model.config.CustomMetricConfig;
+import org.hypertrace.core.documentstore.model.config.postgres.PostgresConnectionConfig;
+import org.hypertrace.core.documentstore.postgres.PostgresDatastore;
+import org.hypertrace.core.documentstore.query.Query;
+import org.hypertrace.core.documentstore.query.SelectionSpec;
+
+@Slf4j
+public class PostgresDocStoreMetricProvider extends BaseDocStoreMetricProviderImpl {
+  private static final String NUM_ACTIVE_CONNECTIONS_METRIC_NAME =
+      "num.active.postgres.connections";
+  private static final String APP_NAME_LABEL = "app_name";
+  private static final String APPLICATION_COLUMN_NAME = "application_name";
+  private static final String PG_STAT_ACTIVITY_TABLE = "pg_stat_activity";
```

Comment:
Do we need to access this as `postgres` user?

Reply:
Will be able to as long as necessary permissions are present. Just want to avoid creating a different connection for this purpose.